tests/instruments/ametek/test_ametek7270.py

```diff
@@ -0,0 +1,508 @@
+import pytest
+
+from pymeasure.test import expected_protocol
+from pymeasure.instruments.ametek import Ametek7270
+
+
+def test_init():
+    with expected_protocol(
+            Ametek7270,
+            [],
+    ):
+        pass  # Verify the expected communication.
+
+
+@pytest.mark.parametrize("comm_pairs, value", (
+    ([(b'SEN', b'27\n')],
+     1.0),
+    ([(b'SEN', b'27\n')],
+     1.0),
+))
+def test_sensitivity_getter(comm_pairs, value):
+    with expected_protocol(
+            Ametek7270,
+            comm_pairs,
+    ) as inst:
+        assert inst.sensitivity == value
+
+
+def test_slope_getter():
+    with expected_protocol(
+            Ametek7270,
+            [(b'SLOPE', b'1\n')],
+    ) as inst:
+        assert inst.slope == 12
+
+
+def test_time_constant_getter():
+    with expected_protocol(
+            Ametek7270,
+            [(b'TC', b'30\n')],
+    ) as inst:
+        assert inst.time_constant == 100000.0
+
+
+def test_x_getter():
+    with expected_protocol(
+            Ametek7270,
+            [(b'X.', b'0.0E+00\n')],
+    ) as inst:
+        assert inst.x == 0.0
+
+
+def test_y_getter():
+    with expected_protocol(
+            Ametek7270,
+            [(b'Y.', b'0.0E+00\n')],
+    ) as inst:
+        assert inst.y == 0.0
+
+
+def test_xy_getter():
+    with expected_protocol(
+            Ametek7270,
+            [(b'XY.', b'0.0E+00,0.0E+00\n')],
+    ) as inst:
+        assert inst.xy == [0.0, 0.0]
+
+
+def test_mag_getter():
+    with expected_protocol(
+            Ametek7270,
+            [(b'MAG.', b'0.0E+00\n')],
+    ) as inst:
+        assert inst.mag == 0.0
+
+
+def test_theta_getter():
+    with expected_protocol(
+            Ametek7270,
+            [(b'PHA.', b'-1.8E+02\n')],
+    ) as inst:
+        assert inst.theta == -180.0
+
+
+@pytest.mark.parametrize("method, command", [('x1', 'X1.'),
+                                             ('y1', 'Y1.'),
+                                             ('x2', 'X2.'),
+                                             ('y2', 'Y2.')])
+def test_failing_properties(method, command):
+    """in standard single reference mode, these tests should raise a ValueError"""
+    with pytest.raises(ValueError):
+        with expected_protocol(
+                Ametek7270,
+                [(f'{command}'.encode(), b'\n')]
+        ) as inst:
+            getattr(inst, method) == 0.0
+
+
+@pytest.mark.parametrize("comm_pairs, value", (
+    ([(b'REFN', b'7\n')],
+     7.0),
+    ([(b'REFN', b'7\n')],
+     7.0),
+    ([(b'REFN', b'7\n')],
+     7.0),
+    ([(b'REFN', b'7\n')],
+     7.0),
+    ([(b'REFN', b'7\n')],
+     7.0),
+    ([(b'REFN', b'7\n')],
+     7.0),
```

Comment:
Why do you run the same test six times? What do you expect to change between these tests?

Reply:
well it was autogenerated from your auto code...I'll change that

Reply:
if you execute one property several times, it accumulates it. The idea is, that you may test different values.